File: fapi/src/main/java/com/netcracker/chargingservice/fapi/models/UserProfile.java
package com.netcracker.chargingservice.fapi.models;

public class UserProfile {

    private Long id;
    private String email;
    private Byte isBlocked;
    private String firstName;
    private String lastName;
    private Role role;

    public UserProfile() {}

    public UserProfile(Long id, String firstName, String lastName, String email, Role role, Byte isBlocked) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.role = role;
        this.isBlocked = isBlocked;
    }

    public Byte getIsBlocked() {
        return isBlocked;
    }

    public void setIsBlocked(Byte isBlocked) {
        this.isBlocked = isBlocked;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Role getRole() {
        return role;
    }

    public void setRole(Role role) {
        this.role = role;
    }
}

File: frontend/src/app/models/user-profile.ts
import {Role} from "./role";

export class UserProfile {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  password: string;
  isBlocked: string;
  role: Role;

  static cloneBase(userProfile: UserProfile): UserProfile {
    const clonedUserProfile: UserProfile = new UserProfile();
    clonedUserProfile.id = userProfile.id;
    clonedUserProfile.email = userProfile.email;
    clonedUserProfile.firstName = userProfile.firstName;
    clonedUserProfile.lastName = userProfile.lastName;
    clonedUserProfile.password = userProfile.password;
    clonedUserProfile.role = userProfile.role;
    clonedUserProfile.isBlocked = userProfile.isBlocked;
    return clonedUserProfile;
  }
}

export class AuthToken {
  token: string;
}
